docs/data/data-grid/localization/localization.md

```diff
@@ -17,6 +17,22 @@ In the following example, the labels of the density selector are customized.
 
 {{"demo": "CustomLocaleTextGrid.js", "bg": "inline"}}
 
+> ⚠️ It's important to note that because the data grid uses components from the Material UI Core library some translation keys need to be accessed using that component key.
+>
+> One example is the `TablePagination` component used in the grid footer when pagination is enabled.
```

Comment:
Would this help? It wasn't clear to me:

```suggestion
> One example is the table pagination component used in the grid footer when pagination is enabled. All the keys provided to the `MuiTablePagination` object are applied as props directly to the [`TablePagination`](/material-ui/api/table-pagination/) component.
```

Reply:
There are 2 problems raised in this issue. The first is the one you pointed out. The second is regarding how to provide your own string so you overwrite the OTB text we have. This PR is related to the second problem. The thing is that we don't explain how to overwrite texts in MUI Core components we use.
It's my mistake that I marked that this PR fixed the linked issue. Its related to what has been discussed there.